src/stores/auth.js

```diff
@@ -0,0 +1,73 @@
+import { defineStore } from "pinia";
+import * as authApi from "@/api/services/Authentication.js";
+import { removeAuthHeader, setAuthHeader } from "@/api/axiosConfig.js";
+
+const TOKEN_STORAGE_KEY = "auth-token";
+const USER_STORAGE_KEY = "user";
+
+export const useAuthStore = defineStore({
+  id: "auth",
+  state: () => ({
+    userData: null,
+    authToken: null,
+  }),
+  getters: {
+    isLoggedIn: (state) => !!state.authToken,
+    isUserLoggedIn: (state) => {
+      return state.authToken ? state.userData.role === "user" : "";
+    },
+    isEmployeeLoggedIn: (state) => {
+      return state.authToken ? state.userData.role === "employee" : "";
+    },
+  },
+  actions: {
+    setUserData({ authToken, user }) {
+      this.authToken = authToken;
+      this.userData = user;
+      localStorage.setItem(TOKEN_STORAGE_KEY, authToken);
+      localStorage.setItem(USER_STORAGE_KEY, JSON.stringify(user));
+    },
+    resetUserData() {
+      this.authToken = null;
+      this.userData = null;
+      localStorage.removeItem(TOKEN_STORAGE_KEY);
+      localStorage.removeItem(USER_STORAGE_KEY);
+    },
+    restoreUserData() {
+      try {
+        const authToken = localStorage.getItem(TOKEN_STORAGE_KEY);
+        const user = JSON.parse(localStorage.getItem(USER_STORAGE_KEY) || null);
+        if (authToken && user) {
```

Comment:
If in local storage is no `USER STORAGE KEY` will it not return null by default? Is there a need to do it explicitly?

Reply:
This solution was recommended to us by Artur: https://github.com/rossinek/vue3-auth-playground/blob/main/src/store/auth.js

Reply:
Sure, Artur is a great developer but sometimes it is good to analyze code on your own even if it works. This is a great way to learn. The restore approach is 100% correct, but what about this particular statement? Paying attention to details will help you write cleaner code

Reply:
I just conclude it was made for a reason to handle some edge cases and I trust his years long experience, since I have no experience in handling authentication using tokens and storing them in local storage.

Reply:
It would be good to know what edge case is that, don't you think?